src/components/input/TextComponent.tsx

```diff
@@ -146,30 +160,37 @@ const TextComponent: React.FC<TextComponentPropsWithType> = ({
           style={
             {
               WebkitTextSecurity:
-                type == Types.TEXTFIELD && masked && hideText && 'disc',
+                type == Types.TEXTFIELD &&
+                (isMasked || (masked && hideText)) &&
+                'disc',
             } as React.CSSProperties
           }
           disabled={disabled}
           {...rest}
         />
-        {iconElement && type == Types.TEXTFIELD
-        // Clone the iconElement in order to attach className 'tk-input__icon'
-          ? React.cloneElement(iconElement, {
-            className: classNames(
-              'tk-input__icon',
-              iconElement.props.className
-            ),
-          })
+
+        {decoratorElement && type == Types.TEXTFIELD
+          ? // Clone the decoratorElement in order to attach className 'tk-input__hide'
+            React.cloneElement(decoratorElement)
           : null}
-        {type == Types.TEXTFIELD && masked && value?.length &&  (
+        {type == Types.TEXTFIELD && masked && value?.length ? (
           <button
             className="tk-input__hide"
```

Comment:
forget to check, but we shouldn't document this class anymore in styles. 
If it's already the case then ignore my comment ;)

Reply:
We have an example that document this class on storybook styles, I can remove it